bbb/meta-bbb/recipes-core/trebletrouble/files/src/tone.c

```diff
@@ -0,0 +1,255 @@
+      wave->data[ wave->index + 0 ] = sample[0];
+      wave->data[ wave->index + 1 ] = sample[1];
+      wave->data[ wave->index + 2 ] = sample[2];
+      wave->data[ wave->index + 3 ] = sample[3];
+      wave->index += 4;
+    }
+  }
+}
+
+void waveToFile( Wave* wave, const char* filename ) {
+
+  /* First make sure all numbers are little endian */
+  toLittleEndian(sizeof(int), (void*)&(wave->header.chunkSize));
+  toLittleEndian(sizeof(int), (void*)&(wave->header.subChunk1Size));
+  toLittleEndian(sizeof(short int), (void*)&(wave->header.audioFormat));
+  toLittleEndian(sizeof(short int), (void*)&(wave->header.numChannels));
+  toLittleEndian(sizeof(int), (void*)&(wave->header.sampleRate));
+  toLittleEndian(sizeof(int), (void*)&(wave->header.byteRate));
+  toLittleEndian(sizeof(short int), (void*)&(wave->header.blockAlign));
+  toLittleEndian(sizeof(short int), (void*)&(wave->header.bitsPerSample));
+  toLittleEndian(sizeof(int), (void*)&(wave->header.subChunk2Size));
+
+  /* Open the file, write header, write data */
+  FILE *file;
+  file = fopen(filename, "wb");
+  fwrite( &(wave->header), sizeof(WaveHeader), 1, file);
+  fwrite( (void*)(wave->data), sizeof(char), wave->size, file);
+  fclose( file);
+  
+  /* Convert back to the system endian-ness */
+  toLittleEndian(sizeof(int), (void*)&(wave->header.chunkSize));
+  toLittleEndian(sizeof(int), (void*)&(wave->header.subChunk1Size));
+  toLittleEndian(sizeof(short int), (void*)&(wave->header.audioFormat));
+  toLittleEndian(sizeof(short int), (void*)&(wave->header.numChannels));
+  toLittleEndian(sizeof(int), (void*)&(wave->header.sampleRate));
+  toLittleEndian(sizeof(int), (void*)&(wave->header.byteRate));
+  toLittleEndian(sizeof(short int), (void*)&(wave->header.blockAlign));
+  toLittleEndian(sizeof(short int), (void*)&(wave->header.bitsPerSample));
+  toLittleEndian(sizeof(int), (void*)&(wave->header.subChunk2Size));
+}
+
+/* -------------------------------------------------------------------- [ Section: Main ]  */
+/* Wave tone(void){ */
+void tone(float freq, float* data, float duration){ 
+  /* Define some variables for the sound */
+  /*float duration = 0.3;*/       /* seconds */
+
+  int nSamples = (int)(duration*SAMPLE_RATE);
+
+  /* Create a mono(1), 32-bit sound and set the duration */
+  Wave mySound = makeWave((int)SAMPLE_RATE,1,32);
+  waveSetDuration(&mySound, duration);
+
+  /* Add all of the data */
+  int i;
+  float frameData[1];
+  for (i = 0; i < nSamples; i++) {
+    data[i] = cos(freq*(float)i*3.14159/SAMPLE_RATE);
+    /*frameData[i] = cos(freq*(float)i*3.14159/SAMPLE_RATE);*/
+  }
+
+  for(i = 0; i<nSamples; i++){
+    frameData[0] = cos(freq*(float)i*3.14159/SAMPLE_RATE);
+    waveAddSample(&mySound,frameData);
+  }
+
+  /* Write it to a file and clear up when done */
+  /* waveToFile( &mySound, "mono-32bit.wav"); */
+  
+  /* destroy the wave and return the sound   */
+
+  /* waveDestroy( &mySound ); */
+
+  /* return mySound; */
+  /* return data; */
```

Comment:
What is going on in this section of commented out code?
This should (most likely) be removed or (unlikely) be uncommented.